src/compas_rhino/interop/__init__.py

```diff
@@ -0,0 +1,175 @@
+"""
+********************************************************************************
+interop
+********************************************************************************
+
+.. currentmodule:: compas_rhino.interop
+
+.. rst-class:: lead
+
+Conversions between Rhino geometry objects (:mod:`Rhino.Geometry`) and COMPAS geometry objects (:mod:`compas.geometry`).
+
+
+Primitives
+==========
+
+.. autosummary::
+    :toctree: generated/
+    :nosignatures:
+
+    compas_point_to_rhino_point
```

Comment:
the return of the non object-oriented API? ;)

I think this is confusing in a few ways:
 - for devs, the link between what objects exists and which one have implemented conversions is hard to follow, might lead to incomplete API
 - the names are too verbose, except a few exceptions (plane-to-frame, and some curves to different types of curves), most of these are from compas_X to rhino_X. I don't think the exceptions really justify the verbosity of all methods
 - there should be one way of doing things, and now there's at least 2 ways of each conversion. And the more worring part is that they don't share the code: many objects have `to_compas` which does the `rhino_bla_to_compas_bla` side of the conversion, and some conversions to rhino geometry exist, for instance, in `compas_ghpython`, doing the other side. If there are multiple ways of doing the same, they should definitely invoke the same code.

Reply:
okay, i will move this to the geometry objects. might create a breaking change here and there though...

Reply:
before i remove the function-based conversion though, how do we feel about the following?

1. the RhinoGeometry wrappers as they currently exist are a bit of a mixed bag: they are not strictly geometry wrappers in the sense that they can be instantiated from doc objects, selections (also doc objects), Rhino geometry, and COMPAS geometry
2. although they can be used to convert COMPAS geometry to Rhino Geometry, the process is not immediately obvious (`rhino_circle = RhinoCircle.from_geometry(compas_circle).geometry`)
3. although there is a `from_object` method, there is no `to_object` method (use cases would be very confusing since one would have to add the geometry object to the doc objects table, even though the object could have been created from a doc object in the first place)
4. to be compatible with the data framework, we would have to derive them from `compas.data.Data`, and make the data dicts compatible with the data dicts of the corresponding COMPAS objects

there is however the advantage of being able to add additional methods such as `closest_point` which allows the Rhino objects to be used as constraints in optimisation processes, without having to check for the type of object at every iteration.

there is also the possibility to add some of the Rhino user data to the data dict representation such that this can be passed to CPython via RPC. however, there is currently no corresponding data attribute on the geometry objects on the COMPAS side

Reply:
@gonzalocasas @beverlylytle